### PR TITLE
Staging

### PR DESCRIPTION
**Pull request by rakshi00**
Add withdraw
Add refresh after status change

### CODE REVIEW

File: frontend/admin/src/pages/purchase/PurchaseDetail.tsx
Comment (ntruessel) [inline frontend/admin/src/pages/purchase/PurchaseDetail.tsx]: Why do we need a timeout to refresh the page?

**Review by ntruessel — COMMENTED** [review]
Looks innocent enough to me, but I would wait for a second review before merging

File: frontend/admin/src/pages/purchase/PurchaseDetail.tsx
Comment (rakshi00) [reply frontend/admin/src/pages/purchase/PurchaseDetail.tsx]: problem is that the toast cannot be displayed after refresh. scripts are reloaded too.
Therefore, the workaround is to wait 1.5s, so that the toast can be seen before refresh for now

Comment (ntruessel) [reply frontend/admin/src/pages/purchase/PurchaseDetail.tsx]: Ok, but then the implementation is wrong. We were able to navigate between different pages using `history.push` without toasts getting lost.

**Review by ntruessel — CHANGES_REQUESTED** [review]
We have to find a way to reload the page without actually reloading everything. I have some ideas (fetching the purchase from the backend again, replacing the current route twice to fool react, ...) but that needs to wait until tomorrow.

File: frontend/admin/src/pages/purchase/PurchaseDetail.tsx
Comment (rakshi00) [reply frontend/admin/src/pages/purchase/PurchaseDetail.tsx]: Reverted refresh for the moment, no refresh is being made after status change in current branch

**Comment by rakshi00** [comment]
- revert changes about refresh after status change
- change 2nd level sorting in purchase list to "highest purchase number first"
- add focus to "today" in purchase list if existing

**Review by ntruessel — APPROVED** [review]

**Comment by ntruessel** [comment]
I'd still prefer if someone else also looked at the purchase withdrawal process

File: backend/src/main/java/de/njsm/versusvirus/backend/service/purchase/PurchaseService.java
Comment (F1rst-Unicorn) [inline backend/src/main/java/de/njsm/versusvirus/backend/service/purchase/PurchaseService.java]: Exception text talks about delete in withdraw context

**Review by F1rst-Unicorn — APPROVED** [review]

File: backend/src/main/java/de/njsm/versusvirus/backend/service/purchase/PurchaseService.java
Comment (rakshi00) [reply backend/src/main/java/de/njsm/versusvirus/backend/service/purchase/PurchaseService.java]: Alright, I fixed it :)

**Pull request merged**
